conjure-lib/src/main/java/com/palantir/conjure/java/lib/Bytes.java

```diff
@@ -113,6 +115,32 @@ public static Bytes from(ByteBuffer buffer) {
         return new Bytes(safe);
     }
 
+    /** Constructs a new {@link Bytes} from the provided {@link ByteArrayOutputStream} */
+    public static Bytes from(ByteArrayOutputStream byteStream) {
+        // this returns a new copy already
+        byte[] safe = byteStream.toByteArray();
+
+        return new Bytes(safe);
+    }
+
+    /** Constructs a new {@link Bytes} by reading the provided {@link InputStream} fully */
+    public static Bytes from(InputStream inputStream) throws IOException {
```

Comment:
I don't think this method is a good idea, we don't want to encourage fully buffering InputStreams on heap. I realize sometimes we end up with code that does this, and the implementation you have provided does it better, but I'm worried that by adding this method, people will use it instead of plumbing more efficient streams end to end.

Reply:
I guess that is fair. If we keep the ByteArrayOutputStream version then I imagine others would duplicate this though.